testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TestDeltaLakeSystemTableCompatibility.java

```diff
@@ -92,4 +92,65 @@ public void testTablePropertiesWithTableFeatures()
+        String tableName = "test_table_partitions_with_deletion_vectors_" + randomNameSuffix();
+        onDelta().executeQuery("" +
+                "CREATE TABLE default." + tableName +
+                "(a INT, b INT, c VARCHAR(1)) " +
+                "USING delta " +
+                "PARTITIONED BY (a, c) " +
+                "LOCATION 's3://" + bucketName + "/databricks-compatibility-test-" + tableName + "' " +
+                "TBLPROPERTIES ('delta.enableDeletionVectors' = true)");
+
+        List<Row> expectedBeforeDelete = ImmutableList.of(
+                row("{\"a\":1,\"c\":\"a\"}", 3L, 1347L, "{\"b\":{\"min\":11,\"max\":19,\"null_count\":0}}"),
+                row("{\"a\":1,\"c\":\"d\"}", 1L, 449L, "{\"b\":{\"min\":13,\"max\":13,\"null_count\":0}}"));
+
+        List<Row> expectedAfterFirstDelete = ImmutableList.of(
+                row("{\"a\":1,\"c\":\"a\"}", 2L, 898L, "{\"b\":{\"min\":11,\"max\":17,\"null_count\":0}}"),
+                row("{\"a\":1,\"c\":\"d\"}", 1L, 449L, "{\"b\":{\"min\":13,\"max\":13,\"null_count\":0}}"));
+
+        List<Row> expectedAfterSecondDelete = ImmutableList.of(
+                row("{\"a\":1,\"c\":\"a\"}", 2L, 898L, "{\"b\":{\"min\":11,\"max\":17,\"null_count\":0}}"));
+
+        try {
+            onDelta().executeQuery("INSERT INTO default." + tableName + " VALUES (1, 11, 'a'), (1, 19, 'a'), (1, 17, 'a'), (1, 13, 'd')");
+            assertThat(onTrino().executeQuery(format("SELECT cast(partition as json), file_count, total_size, cast(data as json) FROM delta.default.\"%s$partitions\"", tableName))).containsOnly(expectedBeforeDelete);
+            onDelta().executeQuery("DELETE FROM default." + tableName + " WHERE b = 19");
+            assertThat(onTrino().executeQuery(format("SELECT cast(partition as json), file_count, total_size, cast(data as json) FROM delta.default.\"%s$partitions\"", tableName))).containsOnly(expectedAfterFirstDelete);
+            onDelta().executeQuery("DELETE FROM default." + tableName + " WHERE c = 'd'");
+            assertThat(onTrino().executeQuery(format("SELECT cast(partition as json), file_count, total_size, cast(data as json) FROM delta.default.\"%s$partitions\"", tableName))).containsOnly(expectedAfterSecondDelete);
+        }
+        finally {
+            onDelta().executeQuery("DROP TABLE IF EXISTS default." + tableName);
+        }
+    }
+
+    @Test(groups = {DELTA_LAKE_OSS, PROFILE_SPECIFIC_TESTS})
+    public void testTablePartitionsWithNoColumnStats()
+    {
+        String tableName = "test_table_partitions_with_no_column_stats_" + randomNameSuffix();
+        onDelta().executeQuery("" +
+                "CREATE TABLE default." + tableName +
+                "(a INT, b INT, c VARCHAR(1)) " +
+                "USING delta " +
+                "PARTITIONED BY (a, c) " +
+                "LOCATION 's3://" + bucketName + "/databricks-compatibility-test-" + tableName + "' " +
+                "TBLPROPERTIES ('delta.dataSkippingNumIndexedCols' = 0)");
```

Comment:
Does this property affect null counts? Could you change the following INSERT statement to insert NULL? 

If would be nice to return NULL instead of 0 if the stats doesn't exist.

Reply:
The property applies to `null_counts` as well. I believe the value is populated internally as 0 instead of NULL when  `delta.dataSkippingNumIndexedCols` is enabled.

Reply:
This is the stats Spark generated. The stats doesn't contain null counts. Please fix the implementation. 
```json
{"commitInfo":{"timestamp":1714004633968,"operation":"WRITE","operationParameters":{"mode":"Append","partitionBy":"[]"},"readVersion":0,"isolationLevel":"Serializable","isBlindAppend":true,"operationMetrics":{"numFiles":"5","numOutputRows":"5","numOutputBytes":"2209"},"engineInfo":"Apache-Spark/3.5.0 Delta-Lake/3.1.0","txnId":"0d85f5b5-8288-46db-9e1b-e08915d3579e"}}
{"add":{"path":"a=1/c=varchar_1/part-00000-15e99230-ea53-4cea-b3a0-4f7e7cdae16f.c000.snappy.parquet","partitionValues":{"a":"1","c":"varchar_1"},"size":449,"modificationTime":1714004633000,"dataChange":true,"stats":"{\"numRecords\":1}"}}
{"add":{"path":"a=1/c=varchar_1/part-00001-7cef1675-9764-4f78-b2ef-76eb0cc2de57.c000.snappy.parquet","partitionValues":{"a":"1","c":"varchar_1"},"size":449,"modificationTime":1714004633000,"dataChange":true,"stats":"{\"numRecords\":1}"}}
{"add":{"path":"a=1/c=varchar_1/part-00002-b46ea678-e8e3-401f-8886-06bc2bf56025.c000.snappy.parquet","partitionValues":{"a":"1","c":"varchar_1"},"size":449,"modificationTime":1714004633000,"dataChange":true,"stats":"{\"numRecords\":1}"}}
{"add":{"path":"a=1/c=varchar_2/part-00003-2ebbfb65-bca5-4505-96fc-6ec5800e1efe.c000.snappy.parquet","partitionValues":{"a":"1","c":"varchar_2"},"size":449,"modificationTime":1714004633000,"dataChange":true,"stats":"{\"numRecords\":1}"}}
{"add":{"path":"a=1/c=varchar_3/part-00004-f2206751-70ed-4121-92d4-dcec9c990bc5.c000.snappy.parquet","partitionValues":{"a":"1","c":"varchar_3"},"size":413,"modificationTime":1714004633000,"dataChange":true,"stats":"{\"numRecords\":1}"}}

```